lib/krane/deploy_task.rb

```diff
@@ -238,18 +210,22 @@ def secrets_from_ejson
     def discover_resources
       @logger.info("Discovering resources:")
       resources = []
-      crds_by_kind = cluster_resource_discoverer.crds.group_by(&:kind)
+      crds_grouped = cluster_resource_discoverer.crds.group_by(&:cr_group_kind)
+      group_kinds = @task_config.group_kinds
```

Comment:
Within task config, you get this from an instance of cluster_resource_discoverer. Why not call it directly here? Since that class controls expensive operations, I'd think we should reuse a single instance as much as possible.